src/Avalonia.Controls/IApplicationLifecycle.cs

```diff
@@ -7,14 +7,25 @@ namespace Avalonia.Controls
     /// </summary>
     public interface IApplicationLifecycle
     {
+        /// <summary>
+        /// Sent when the application is starting up.
+        /// </summary>
+        event EventHandler<StartupEventArgs> Startup;
+
         /// <summary>
         /// Sent when the application is exiting.
         /// </summary>
-        event EventHandler OnExit;
+        event EventHandler<ExitEventArgs> Exit;
+
+        /// <summary>
+        /// Shuts down an application that returns the specified exit code to the operating system.
```

Comment:
> that returns the specified exit code to the operating system.

There's no exit code passed to this method, I think this was copy/pasted from the other overload?

Reply:
I guess this just complicated to get right. The exit code that is set on the event args is set to Environment.ExitCode when this is handled.